pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -882,19 +867,49 @@ func verifyStatus(ctx *domainContext, status *types.DomainStatus) {
 		// check if qemu processes has crashed
 		hasQemu := status.VirtualizationMode == types.HVM || status.VirtualizationMode == types.FML || status.IsContainer
 		if configActivate && status.Activated && hasQemu && !hyper.IsDeviceModelAlive(status.DomainId) {
-			errStr := fmt.Sprintf("verifyStatus(%s) qemu crashed",
+			//Have to check on the domain again before confirming qemu crash because of https://www.pivotaltracker.com/story/show/171962732
+			//Here we pass 3 sec wait before we verify if the domain is still alive.
+			//3 sec should be more than enough for the domain to kill qemu and shutdown itself.
+			log.Infof("verifyStatus(%s): qemu crashed or not running. Checking if the domain is still running",
 				status.Key())
-			log.Errorf(errStr)
-			status.LastErr = "qemu crashed - please restart application instance"
-			status.LastErrTime = time.Now()
-			status.Activated = false
-			status.State = types.HALTED
-			publishDomainStatus(ctx, status)
-			if err := hyper.Delete(status.DomainName, status.DomainId); err != nil {
-				log.Errorf("failed to delete domain: %s (%v)\n", status.DomainName, err)
+			_, err := getCurrentDomainID(ctx, configActivate, status, 3*time.Second)
+			if err == nil {
+				errStr := fmt.Sprintf("verifyStatus(%s) qemu crashed",
+					status.Key())
+				log.Errorf(errStr)
+				status.LastErr = "qemu crashed - please restart application instance"
+				status.LastErrTime = time.Now()
+				status.Activated = false
+				status.State = types.HALTED
+				publishDomainStatus(ctx, status)
+				if err := hyper.Delete(status.DomainName, status.DomainId); err != nil {
+					log.Errorf("failed to delete domain: %s (%v)\n", status.DomainName, err)
+				}
+			}
+		}
+	}
+}
+
+func getCurrentDomainID(ctx *domainContext, configActivate bool, domainStatus *types.DomainStatus, waitBeforeCheckSec time.Duration) (int, error) {
+	time.Sleep(waitBeforeCheckSec)
```

Comment:
The unit of that argument is not seconds, but time.Duration (which is currently nanoseconds). So this is misnamed.

Reply:
Changed the variable to `waitBeforeCheck`.